src/builder/create_attachment.rs

```diff
@@ -117,4 +106,178 @@ impl CreateAttachment {
         encoded.insert_str(0, "data:image/png;base64,");
         encoded
     }
+
+    /// Sets a description for the file (max 1024 characters).
+    pub fn description(mut self, description: impl Into<String>) -> Self {
+        self.description = Some(description.into());
+        self
+    }
+}
+
+#[derive(Debug, Clone, serde::Serialize)]
+struct ExistingAttachment {
+    id: AttachmentId,
+}
+
+#[derive(Debug, Clone, serde::Serialize)]
+#[serde(untagged)]
+enum NewOrExisting {
+    New(CreateAttachment),
+    Existing(ExistingAttachment),
+}
+
+/// You can add new attachments and edit existing ones using this builder.
+///
+/// When this builder is _not_ supplied in a message edit, Discord keeps the attachments intact.
+/// However, as soon as a builder is supplied, Discord removes all attachments from the message. If
+/// you want to keep old attachments, you must specify this either using [`Self::keep_all`], or
+/// individually for each attachment using [`Self::keep`].
+///
+/// # Examples
+///
+/// ## Removing all attachments
+///
+/// ```rust,no_run
+/// # use serenity::all::*;
+/// # async fn _foo(ctx: Context, mut msg: Message) -> Result<(), Error> {
```

Comment:
```suggestion
/// # async fn _foo(ctx: Http, mut msg: Message) -> Result<(), Error> {
```
This fixes the CI failure with no gateway. The other doctests should also be changed in this way.